### PR TITLE
Add icon for IronOS integration

### PR DESCRIPTION
**Pull request by tr4nt0r**
## Type of change

- [x] Add a new logo or icon for a new core integration
- [ ] Add a missing icon or logo for an existing core integration
- [ ] Add a new logo or icon for a custom integration (custom component)
  - [ ] I've added a link to my custom integration repository in the PR description
  - [ ] I've opened up a PR for my custom integration on the [Home Assistant
    Python wheels repository](https://github.com/home-assistant/wheels-custom-integrations)
- [ ] Replace an existing icon or logo with a higher quality version
- [ ] Replace an existing icon or logo after a branding change
- [ ] Removing an icon or logo

## Additional information

- This PR fixes or closes issue: fixes #
- Link to code base pull request: https://github.com/home-assistant/core/pull/120802
- Link to documentation pull request: 
- Link to integration documentation on our website: 
- Link to custom integration repository:

## Checklist

- [ ] The added/replaced image(s) are **PNG**
- [ ] Icon image size is 256x256px (`icon.png`)
- [ ] hDPI icon image size is 512x512px for  (`icon@2x.png`)
- [ ] Logo image size has min 128px, but max 256px, on the shortest side (`logo.png`)
- [ ] hDPI logo image size has min 256px, but max 512px, on the shortest side (`logo@2x.png`)

### CODE REVIEW

File: core_integrations/iron_os/icon.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+mdi:soldering-iron
```
Comment (frenck) [inline core_integrations/iron_os/icon.txt:1]: This is not a helper or a Home Assistant base internal component. This this should not be using this MDI icon.

../Frenck

**Review by frenck — CHANGES_REQUESTED** [review]

File: core_integrations/iron_os/icon.txt (hunk 1 above)
Comment (tr4nt0r) [reply core_integrations/iron_os/icon.txt:1]: Using the MDI icon was a suggestion by @joostlek as IronOS is not a brand and doesn't have a logo. I had suggested this but Joost just laughed 😬

Comment (joostlek) [reply core_integrations/iron_os/icon.txt:1]: Oh, the soldering iron was suggested by Frenck

**Comment by anonymousaga** [comment]
Should we consider using an upscaled version of the optional bootup logo?

Ralim could likely provide vector graphics.

https://github.com/Ralim/IronOS/discussions/1119#discussioncomment-1708997

**Comment by tr4nt0r** [comment]
> Should we consider using an upscaled version of [the optional bootup logo](https://github.com/Ralim/IronOS/discussions/1119#discussioncomment-1708997)?
> 
> Ralim could likely provide vector graphics.
> 
> 

I think it is like the flames thing i posted above, it is a bit cringey, so not sure if that is what we or Ralim would want as logo in HA. But maybe we should ask Ralim if he would like to contribute a logo. Otherwise, i think we go forward with the mdi icon for now until the question about the logo is clarified. Still better than no logo...

**Comment by anonymousaga** [comment]
@tr4nt0r mdi icon will not be allowed. Only home assistant internal integrations get to use mdi logos.

File: core_integrations/iron_os/icon.txt (hunk 1 above)
Comment (frenck) [reply core_integrations/iron_os/icon.txt:1]: What I meant with that, is just picking the raw icon image from their website.
Like this:

light
dark

**Comment by tr4nt0r** [comment]
Then I don't know. What if it is a different color? I don't think it would be right to just make up a logo.

**Comment by frenck** [comment]
> Then I don't know. What if it is a different color? I don't think it would be right to just make up a logo.

Yup exactly :)

**Comment by tr4nt0r** [comment]
Ok, cool 👍🏽
any suggestions or preferences for the color?

**Pull request closed without merging**